src/azimuth/system/resource.c

```diff
@@ -40,6 +42,17 @@ extern const struct resource_entry {
 extern const size_t resource_index_size;
 extern const char _binary_resources_start[];
 
+#if defined(__APPLE__)
+// Use app-bundle resources on macOS
+bool az_system_resource_reader(const char *name, az_reader_t *reader) {
+  const char *resource_dir = SDL_GetBasePath();
```

Comment:
According to https://wiki.libsdl.org/SDL_GetBasePath it seems that the returned string is owned by the caller and needs to be freed with `SDL_free`.